### PR TITLE
Mikefab search admin event

### PR DESCRIPTION
**Pull request by mikefab**

### CODE REVIEW

File: public/view/overlay-controls/admin-search.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    // Return array of admin names
+    }).map(function(key) {
+      return admin_objs[key].name + ' ' +
+      admin_objs[key].admin_code;
+    }).sort();
+    return admins;
+  },
+
+  optionSelected: function() {
+    var argument = arguments[0];
```
Comment (gatoatigrado) [inline public/view/overlay-controls/admin-search.jsx]: can you just make this a regular argument of the function?

File: public/view/overlay-controls/admin-search.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    var admin_objs = this.props.admin_details.admin_data_by_code;
+    var admin_code = Object.keys(this.props.admin_details.admin_data_by_code)
+    .find(function(key) {
+      return admin_objs[key].name + ' ' + admin_objs[key].admin_code === argument;
+    });
+
+    this.props.admin_details.selected_admins.select_admin(admin_code);
+    this.props.admin_details.selected_admins.search_admin(admin_code);
```
Comment (gatoatigrado) [inline public/view/overlay-controls/admin-search.jsx]: huh, you might move the `select_admin(admin_code)` call inside `search_admin()`? doing so seems like "letting the model determine how to mutate its own state", which is nice. what do you think?

File: public/view/overlay-controls/admin-search.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    var admins = Object.keys(this.props.admin_details.admin_data_by_code)
+    .filter(function(key) {
+      var admin_code = admin_objs[key].admin_code.split('-')[0];
+      return available_countries.find(function(e) {
+        return e === admin_code;
+      });
+    // Return array of admin names
+    }).map(function(key) {
+      return admin_objs[key].name + ' ' +
+      admin_objs[key].admin_code;
```
Comment (gatoatigrado) [inline public/view/overlay-controls/admin-search.jsx]: nitty nitty nitpck you don't need to wrap this i think :smile:

File: public/view/overlay-controls/admin-search.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }).sort();
+    return admins;
+  },
+
+  optionSelected: function() {
+    var argument = arguments[0];
+    var admin_objs = this.props.admin_details.admin_data_by_code;
+    var admin_code = Object.keys(this.props.admin_details.admin_data_by_code)
+    .find(function(key) {
+      return admin_objs[key].name + ' ' + admin_objs[key].admin_code === argument;
```
Comment (gatoatigrado) [inline public/view/overlay-controls/admin-search.jsx]: up to you: you could build up a map from search key to admin code, on line 23 above, and just do a lookup here. That way, if you change how something is updated, you can't update one call site but not the other. It's not a big deal though.

File: public/model/selected-admins.js [hunk 5]
```diff
@@ -58,6 +63,26 @@ var SelectedAdmins = P({
...
+    // `searched_admin_codes` accurately reflects what is now searched.
+    var unsearch_cbs = _.values(this.searched_admin_codes);
+    this.searched_admin_codes = {};
+    this.searched_admin_codes[admin_code] = on_unsearch || _.noop;
+    unsearch_cbs.forEach(function(cb) { cb(); });
+    this.selection_ee.emit(
+      new SelectionEvents.AdminSearchEvent(this.get_admin_codes())
+    );
+
+    this.onUpdate();
```
Comment (gatoatigrado) [inline public/model/selected-admins.js]: we're not doing `onUpdate()` anymore, just make the AdminSearchEvent handler in `main.jsx` call `rerender`.

Comment (mikefab) [inline public/model/selected-admins.js]: Done.

File: public/view/overlay-controls/admin-search.jsx (hunk 3 above)
Comment (mikefab) [inline public/view/overlay-controls/admin-search.jsx]: Done.

File: public/view/overlay-controls/admin-search.jsx (hunk 1 above)
Comment (mikefab) [inline public/view/overlay-controls/admin-search.jsx]: Done.

File: public/view/overlay-controls/admin-search.jsx (hunk 4 above)
Comment (mikefab) [inline public/view/overlay-controls/admin-search.jsx]: Done.

File: public/view/overlay-controls/admin-search.jsx (hunk 2 above)
Comment (mikefab) [inline public/view/overlay-controls/admin-search.jsx]: That seems better. Done.

**Pull request merged**